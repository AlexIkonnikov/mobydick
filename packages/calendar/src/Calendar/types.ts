import {MarkingProps} from 'react-native-calendars/src/calendar/day/marking';
import {CalendarProps} from 'react-native-calendars';
import {ReactElement} from 'react';

export interface IMarkedTypes {
  [key: string]: MarkingProps;
}

export interface IMarkedDates {
  dates: IMarkedTypes;
  fromDate: Date;
  toDate: Date;
}
export type colorElem = {
  color: string;
  textColor: string;
};

export type IColors = {
  colorPrime: colorElem;
  colorSoft: colorElem;
};

export interface IDateRange {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}

export interface ICalendar extends CalendarProps {
  onDateRangeChange?: (dateRange: IDateRange) => void;
  bottomView?: ReactElement;
  defaultLocale?: string;
  isClear?: boolean;
  isShowToday?: boolean;
  localeConfig?: ILocaleConfig | undefined;
  isPeriod?: boolean;
}

export interface ILocaleConfig {
  monthNames: string[];
  monthNamesShort: string[];
  dayNames: string[];
  dayNamesShort: string[];
  today?: string;
}
