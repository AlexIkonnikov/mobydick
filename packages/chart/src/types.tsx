import type {ReactElement} from 'react';
import type {SharedValue} from 'react-native-reanimated';

export interface ICoordinates {
  x: number;
  y: number;
}

export interface IChart {
  name?: string;
  coordinates: ICoordinates[];
  colors?: string[];
}

export interface IDataset {
  [key: string]: IChart[];
}

export type IFormatter = (value: number) => string;

export interface IGeneratePeriodsWithPaths {
  dataset: IDataset;
  width: number;
  height: number;
}

export type IRenderSectionItem = (
  {
    period,
    state,
    transition,
  }: {
    period: string;
    state: ISharedGraphState;
    transition: IChartTransition;
  },
  index: number,
) => ReactElement;

export interface IGraphState {
  next: number;
  current: number;
}

export type ISharedGraphState = SharedValue<IGraphState>;
export type IChartTransition = SharedValue<number>;
