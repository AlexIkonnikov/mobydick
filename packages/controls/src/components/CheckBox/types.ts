import React from 'react';

import {IControlProps} from '../types';

interface ICheckBox {
  children: React.ReactNode;
}

export type TCheckBox = ICheckBox & Omit<IControlProps, 'text'>;
