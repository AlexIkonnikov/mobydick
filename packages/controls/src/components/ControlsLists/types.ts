import {FunctionComponentElement} from 'react';
import {PressableProps} from '@npm/mobydick-core';

import {IControlProps} from '../types';

export interface IControlsList {
  onChange(values: string[]): void;
  children: FunctionComponentElement<IControlProps & PressableProps>[];
  horizontal?: boolean;
  single?: boolean;
}
