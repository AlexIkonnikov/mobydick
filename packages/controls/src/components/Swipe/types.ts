export interface ISwipe {
  active: boolean;
  disabled: boolean;
  onPress: (event: boolean) => void;
}
