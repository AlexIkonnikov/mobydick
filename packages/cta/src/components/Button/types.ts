import {ITouchableOpacityProps} from '@npm/mobydick-core';
import {ReactElement} from 'react';
import {StyleProp, TextStyle} from 'react-native';
import {TypographyProp} from '@npm/mobydick-typography';

type IProps = {
  type?: IButtonTypes;
  text?: string;
  textStyle?: StyleProp<TextStyle>;
  font?: TypographyProp | undefined;
  leftIcon?: ReactElement;
  rightIcon?: ReactElement;
  loading?: boolean;
  size?: IButtonSize;
  count?: number;
};

export enum IButtonTypes {
  primary = 'primary',
  secondary = 'secondary',
  tertiary = 'tertiary',
  disabled = 'disabled',
  loading = 'loading',
  destructive = 'destructive',
}

export enum IButtonSize {
  large = 'large',
  small = 'small',
  fixed = 'fixed',
}

export type IButtonProps = ITouchableOpacityProps & IProps;
