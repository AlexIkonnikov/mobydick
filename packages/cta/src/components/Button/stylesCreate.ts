import {IUseStylesTheme} from '@npm/mobydick-styles';
import {StyleSheet, ViewStyle} from 'react-native';

import {ISize, ITypes} from './types';

const sizeButtonStyle = (theme: IUseStylesTheme, size: ISize): ViewStyle => {
  switch (size) {
    case ISize.small:
      return {
        minHeight: theme.spaces.Space32,
        paddingVertical: theme.spaces.Space6,
        paddingHorizontal: theme.spaces.Space16,
      };
    case ISize.large:
      return {
        minHeight: theme.spaces.Space48,
        paddingVertical: theme.spaces.Space12,
        paddingHorizontal: theme.spaces.Space32,
      };
    case ISize.fixed:
      return {
        alignSelf: 'stretch',
        minHeight: theme.spaces.Space48,
      };
  }
};

const primaryStyle = (theme: IUseStylesTheme, size: ISize) => {
  const {colors} = theme;

  return StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: colors.CtaBtnPrimary,
      borderRadius: theme.spaces.Space12,

      ...sizeButtonStyle(theme, size),
    },

    text: {
      color: colors.TextWhite,
    },
    leftIcon: {
      paddingRight:
        size === ISize.small ? theme.spaces.Space4 : theme.spaces.Space6,
    },
    rightIcon: {
      paddingLeft:
        size === ISize.small ? theme.spaces.Space4 : theme.spaces.Space6,
    },
  });
};

const secondaryStyle = (theme: IUseStylesTheme, size: ISize) => {
  const defaultStyles = primaryStyle(theme, size);
  const {
    colors: {CtaBtnSecondary, TextAccent},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnSecondary;
  defaultStyles.text.color = TextAccent;

  return defaultStyles;
};

const tertiaryStyle = (theme: IUseStylesTheme, size: ISize) => {
  const defaultStyles = primaryStyle(theme, size);
  const {
    colors: {TextAccent},
  } = theme;

  defaultStyles.container.backgroundColor = 'transparent';
  defaultStyles.text.color = TextAccent;

  return defaultStyles;
};

const disabledStyle = (theme: IUseStylesTheme, size: ISize) => {
  const defaultStyles = primaryStyle(theme, size);
  const {
    colors: {CtaBtnMuted, TextWhite},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnMuted;
  defaultStyles.text.color = TextWhite;

  return defaultStyles;
};

const destructiveStyle = (theme: IUseStylesTheme, size: ISize) => {
  const defaultStyles = primaryStyle(theme, size);
  const {
    colors: {CtaBtnDestructive, TextWhite},
  } = theme;

  defaultStyles.container.backgroundColor = CtaBtnDestructive;
  defaultStyles.text.color = TextWhite;

  return defaultStyles;
};

const stylesCreate = (theme: IUseStylesTheme, type: ITypes, size: ISize) => {
  switch (type) {
    case ITypes.secondary:
      return secondaryStyle(theme, size);
    case ITypes.tertiary:
      return tertiaryStyle(theme, size);
    case ITypes.disabled:
      return disabledStyle(theme, size);
    case ITypes.destructive:
      return destructiveStyle(theme, size);
    case ITypes.primary:
    default:
      return primaryStyle(theme, size);
  }
};
export default stylesCreate;
