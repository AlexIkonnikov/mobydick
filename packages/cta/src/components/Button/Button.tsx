import {Text, TouchableOpacity} from '@mobydick/core';
import React, {FC} from 'react';
import {Spinner} from '@mobydick/progress';
import {useStyles} from '@mobydick/styles';

import {ButtonProps, ISize, ITypes} from './types';
import stylesCreate from './stylesCreate';

const Button: FC<ButtonProps> = props => {
  const {
    style,
    text,
    leftIcon,
    rightIcon,
    size = ISize.fixed,
    type = ITypes.primary,
    loading = false,
    disabled = false,
    ...otherProps
  } = props;
  const [styles] = useStyles(
    stylesCreate,
    disabled ? ITypes.disabled : type,
    size,
  );

  const Container: FC = ({children}) => (
    <TouchableOpacity
      style={[styles.container, style]}
      disabled={
        loading ||
        disabled ||
        type === ITypes.disabled ||
        type === ITypes.loading
      }
      {...otherProps}>
      {children}
    </TouchableOpacity>
  );

  if (loading || type === ITypes.loading)
    return (
      <Container>
        <Spinner />
      </Container>
    );

  return (
    <Container>
      {leftIcon}
      {Boolean(text) && <Text style={styles.text}>{text}</Text>}
      {rightIcon}
    </Container>
  );
};
export default Button;
