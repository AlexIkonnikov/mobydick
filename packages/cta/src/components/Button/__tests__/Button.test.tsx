import React from 'react';
import {render} from '@testing-library/react-native';
import {View} from '@npm/mobydick-core';

import {ISize, ITypes} from '../types';
import Button from '../index';

describe('@npm/mobydick-cta/Button', () => {
  it('renders correctly', () => {
    const {toJSON, rerender} = render(<Button text={'text'} />);
    expect(toJSON()).toMatchSnapshot();

    rerender(<Button text={'text'} font={'Bold-Error-L'} size={ISize.small} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly secondary', () => {
    const {toJSON} = render(<Button type={ITypes.secondary} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly tertiary', () => {
    const {toJSON} = render(<Button type={ITypes.tertiary} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly disabled', () => {
    const {toJSON} = render(<Button type={ITypes.disabled} disabled={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly destructive', () => {
    const {toJSON} = render(
      <Button type={ITypes.destructive} size={ISize.fixed} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly loading', () => {
    const {toJSON} = render(<Button type={ITypes.loading} loading={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly fixed', () => {
    const {toJSON} = render(<Button size={ISize.fixed} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly small', () => {
    const {toJSON, rerender} = render(<Button size={ISize.small} />);
    expect(toJSON()).toMatchSnapshot();

    rerender(<Button size={ISize.small} loading={true} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly large', () => {
    const {toJSON} = render(<Button size={ISize.large} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly left icon size large', () => {
    const {toJSON} = render(
      <Button
        size={ISize.large}
        leftIcon={<View />}
        rightIcon={<View />}
        text={'text'}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly left icon small', () => {
    const {toJSON} = render(
      <Button size={ISize.small} leftIcon={<View />} rightIcon={<View />} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly without size', () => {
    const {toJSON} = render(<Button size={'ISize' as unknown as ISize} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
