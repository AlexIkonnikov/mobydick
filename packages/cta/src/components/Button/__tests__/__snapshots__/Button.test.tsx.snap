// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-cta/Button renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 18,
          "lineHeight": 22,
        },
        Array [
          Object {
            "color": "#FFF",
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;

exports[`@npm/mobydick-cta/Button renders correctly 2`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#F54D3D",
          "fontFamily": "Inter-Bold",
          "fontSize": 18,
          "lineHeight": 22,
        },
        Array [
          Object {
            "color": "#FFF",
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;

exports[`@npm/mobydick-cta/Button renders correctly disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#9BA1B0",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
/>
`;

exports[`@npm/mobydick-cta/Button renders correctly fixed 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
/>
`;

exports[`@npm/mobydick-cta/Button renders correctly large 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": undefined,
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
/>
`;

exports[`@npm/mobydick-cta/Button renders correctly loading 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <SvgMock
      fill="#FFF"
      height="24"
      width="24"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-cta/Button renders correctly secondary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "rgba(43, 120, 238, 0.15)",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
/>
`;

exports[`@npm/mobydick-cta/Button renders correctly small 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": undefined,
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 32,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 6,
    }
  }
/>
`;

exports[`@npm/mobydick-cta/Button renders correctly small 2`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": undefined,
      "backgroundColor": "#2B78EE",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 32,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 6,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "rotate": "0deg",
          },
        ],
      }
    }
  >
    <SvgMock
      fill="#FFF"
      height="20"
      width="20"
    />
  </View>
</View>
`;

exports[`@npm/mobydick-cta/Button renders correctly tertiary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "transparent",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
/>
`;
